Bottom: e194ebe7403dac663ec411ec39b09701687752c7
Top:    adfe8272d8b30eb010debdd35514a1389f3984bc
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-23 20:56:22 +0000

find.c: Fix inverted logic in LookupConnection()

When AndRats is true, we may consider starting with a ratline object,
otherwise we do no. I got this backwards by mistake.


---

diff --git a/src/find.c b/src/find.c
index a1d6bce..3622777 100644
--- a/src/find.c
+++ b/src/find.c
@@ -3067,7 +3067,7 @@ LookupConnection (Coord X, Coord Y, bool AndDraw, Coord Range, int which_flag,
   if (type == NO_TYPE)
     {
       type = SearchObjectByLocation (
-        LOOKUP_MORE & ~(AndRats ? RATLINE_TYPE : 0),
+        LOOKUP_MORE & ~(AndRats ? 0 : RATLINE_TYPE),
         &ptr1, &ptr2, &ptr3, X, Y, Range);
       if (type == NO_TYPE)
         return;
